gprofiler/profilers/java.py

```diff
@@ -698,8 +707,11 @@ def __init__(
         self._metadata = JavaMetadata(self._stop_event)
         self._insert_dso_name = insert_dso_name
 
-    def _init_ap_mode(self, ap_mode: str) -> None:
-        if ap_mode == "auto":
+    def _init_ap_mode(self, profiling_mode: str, ap_mode: str) -> None:
+        if profiling_mode == "allocation":
```

Comment:
`assert profiling_mode in ("cpu", "allocation")` as well

Reply:
Checked in the "framework", is there any reason why I should do a double-check?

Reply:
This is not a check this is an assert, it clarifies for the reader what the function expects to handle. I like adding many asserts

Reply:
kk, done